server/cmwell-data-tools/src/main/scala/cmwell/tools/data/downloader/consumer/BufferFillerActor.scala

```diff
@@ -142,13 +144,20 @@ class BufferFillerActor(threshold: Int,
       tsvCounter += 1
 
     case GetData if buf.nonEmpty =>
-      sender ! buf.dequeue()
+     sender ! ( buf.dequeue() match {
+        case None => None
+        case Some(tokenAndData) => Some(tokenAndData._1,tokenAndData._2,horizon)
+      } )
```

Comment:
BTW, it will be nicer to write:
`buf.dequeue().map(tokenAndData => Some(tokenAndData._1,tokenAndData._2,horizon))` which will result the same